modules/spring-config/src/main/java/org/mule/runtime/config/spring/Notification.java

```diff
@@ -0,0 +1,234 @@
+import org.mule.runtime.core.api.context.notification.ConnectionNotificationListener;
+import org.mule.runtime.core.api.context.notification.ConnectorMessageNotificationListener;
+import org.mule.runtime.core.api.context.notification.CustomNotificationListener;
+import org.mule.runtime.core.api.context.notification.ExceptionNotificationListener;
+import org.mule.runtime.core.api.context.notification.ExceptionStrategyNotificationListener;
+import org.mule.runtime.core.api.context.notification.ManagementNotificationListener;
+import org.mule.runtime.core.api.context.notification.MessageProcessorNotificationListener;
+import org.mule.runtime.core.api.context.notification.MuleContextNotificationListener;
+import org.mule.runtime.core.api.context.notification.RegistryNotificationListener;
+import org.mule.runtime.core.api.context.notification.RoutingNotificationListener;
+import org.mule.runtime.core.api.context.notification.SecurityNotificationListener;
+import org.mule.runtime.core.api.context.notification.TransactionNotificationListener;
+import org.mule.runtime.core.context.notification.AsyncMessageNotification;
+import org.mule.runtime.core.context.notification.ComponentMessageNotification;
+import org.mule.runtime.core.context.notification.ConnectionNotification;
+import org.mule.runtime.core.context.notification.ConnectorMessageNotification;
+import org.mule.runtime.core.context.notification.CustomNotification;
+import org.mule.runtime.core.context.notification.ExceptionNotification;
+import org.mule.runtime.core.context.notification.ExceptionStrategyNotification;
+import org.mule.runtime.core.context.notification.ManagementNotification;
+import org.mule.runtime.core.context.notification.MessageProcessorNotification;
+import org.mule.runtime.core.context.notification.MuleContextNotification;
+import org.mule.runtime.core.context.notification.PipelineMessageNotification;
+import org.mule.runtime.core.context.notification.RegistryNotification;
+import org.mule.runtime.core.context.notification.RoutingNotification;
+import org.mule.runtime.core.context.notification.SecurityNotification;
+import org.mule.runtime.core.context.notification.TransactionNotification;
+import org.mule.runtime.core.util.ClassUtils;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * Notification helper object for creating notification manager configuration.
+ *
+ * For now we create specific types {@code Notification.EnabledNotification} and {@code Notification.DisabledNotification}
+ * for elements that have children of the same type but that must be injected in different collections of the
+ * parent object. Ideally this should not be needed or we don't have to support such use case anymore but for
+ * previous elements coming from previous mule version we have to keep the XML syntax as it is.
+ *
+ * @since 4.0
+ */
+public class Notification
```

Comment:
this class name is not descriptive. The Jdoc is also confusing since it says that it's a helper for creating a manager configuration, yet it doesn't look like a factory or builder, it rather looks like an actual notification